### ISSUE TITLE
Little Snitch blocking VSCode (PlatformIO) due to "modification of the program"

### ISSUE THREAD

**Issue by demosthien** [microsoft/vscode#77344]
This bug was originally reported to the PlatformIO Github page but their response was to tell me to report it to vscode. See the link below for full details including updates and further troubleshooting to find the cause of the issue;

https://github.com/platformio/platformio-vscode-ide/issues/809#issuecomment-511133485

Following is the first part of my posted issue report.

==========

> Operating System: OSX 10.14.3
> 
> Just an FYI for the devs and anyone else who may come upon this issue (I see it seems something similar happened a couple of times in 2018).
> 
> I had previously (about 4 months ago) installed VSCode and PlatformIO (PIO) extension but had shortly thereafter uninstalled.
> 
> I recently reinstalled VSCode but when adding PIO it would return an error and not complete the installation. Unfortunately the error returned by VSCode, and presented to "Report" (by directly creating an issue here on Github) was in URL code. I've seen a few of these show up recently so that might be a seperate something the devs should also look at.
> 
> After playing around with uninstalling and reinstalling VSCode and PIO (including finding the hidden Users/XXXXX/.platformio/ folder and deleting it) it still returned the same error.
> 
> In the end, after scouring the issues here, I found a couple of mentions of Little Snitch blocking VSCode. Sure enough, on checking my Little Snitch it was blocking VSCode showing the problem as;
> 

> > The identity check detected a modification of the program. Therefore all of its connections were denied as a precaution.

> After allowing VSCode once again PIO has installed correctly. However, I suspect that Little Snitch will block VSCode again should it detect any future modifications to the app by PIO?
> 
> Below are three of the error reports I got during this process (the first entry below was riddled with URL code so I [decoded](https://meyerweb.com/eric/tools/dencoder/) it for clarity but that also likely removed some of the legit code within the text block the following to were copied direct from the VSCode error message), they're probably the same (they look it), but I've pasted them all here just in case there's a difference I don't see.
> 
> Hopefully this will help anyone encountering this issue going forward.

**Comment by joaomoreno** [comment]
> I recently reinstalled VSCode but when adding PIO it would return an error 

What error?

**Comment by demosthien** [comment]
All the details I have are provided in the linked Issue submitted to PIO.

**Comment by joaomoreno** [comment]
All the errors in that issue are PIO errors.

Is VS Code returning any error?

**Comment by demosthien** [comment]
Sorry mate, I've got nothing else I can add for that particular set of occasions. I believe the errors thrown up by PIO were only because it was trying to do some kind of online handshake/download at the time, which wouldn't pass Little Twitch because VSC/PIO had changed some file or something. VSC was happy to keep working so probably registered no error as it wasn't trying to do the online connection. ¯\_(ツ)_/¯ But I don't know enough about these things to know for sure.

VSC does still have the same issue with my Little Twitch firewall though as I've got to turn it off to use VSC... Is there something else I can do to get more info for you because I can try to replicate the problem, is there an error or issue log file I could post?

**Comment by joaomoreno** [comment]
I still don't understand what happened, as the issue description is too confusing. Let's start from the beginning. You start VS Code. And then... what doesn't work?

**Comment by demosthien** [comment]
Right...

Short version; VSC is being refused internet access by the firewall Little Snitch, claiming modification of program has occurred, and this causes issues when VSC can't download stuff.

Longer version; When VSC (and/or I assume an extension such as PIO in my case) attempted to connect to some internet location and my firewall (Little Snitch) blocks the connection.

On investigation Little Snitch appeared to be blocking the connection and reports the reason as;

> The identity check detected a modification of the program. Therefore all of its connections were denied as a precaution.

This refused connection prevents VSC/PIO connecting to whatever it was looking for and PIO would return an error message (there is also a problem tied up with my other report where PIO's auto error reporting system where it was generating messages in URL code but that isn't an issue for VSC). A selection of these PIO error messages are shown in the report I submitted to the PIO repo (see link in OP). The refused connection would then cause follow-up issues because VSC/PIO could not download whatever it was after.

I have also included some information provided by Little Snitch, regarding why it blocked the VSC internet connections (see the PIO report for details).

On further investigation, turning off Little Snitch permitted all the things to work fine. But obviously turning off the firewall is not a good long-term solution.

I initially report the error to the PIO repo but they referred me to you.
